microya/src/androidTest/java/com/papershift/microya/MicroyaIntegrationTests.kt

```diff
@@ -184,7 +228,61 @@ class MicroyaIntegrationTests {
     @Test(expected = IllegalArgumentException::class)
     fun testMockedWithCustomResponseDeleteRequest() {
         runBlocking {
-            mockedWithCustomResponseApiProvider.performRequest<FooBar, PostmanEchoError>(PostmanEchoEndpoint.Delete).getError()
+            mockedWithCustomResponseApiProvider.performRequest<FooBar, PostmanEchoError>(
+                PostmanEchoEndpoint.Delete
+            ).getError()
         }
     }
-}
\ No newline at end of file
+
+    @Test
+    fun testUploadRequest() {
+        runBlocking {
+            val fileDataPart = FileDataPart(file = getImage(), name = "image")
+            val uploadImageRequest = UploadImageRequest(
+                "japan", "japan summer"
+            )
+            val uploadEndpoint = ImgurEndpoint.UploadImageEndpoint(
+                uploadImageRequest
+            )
+            val result: ImgurResponse<UploadSuccessResponse> =
+                uploadFileSampleApiProvider.performUploadRequest<ImgurResponse<UploadSuccessResponse>, ImgurResponse<ImgurErrorResponse>>(
+                    uploadEndpoint, listOf(fileDataPart)
+                ).get()!!
```

Comment:
From the usage side it looks weird to expect a `Response` wrapped in a `Response`. I also don't understand why I have to specify two times the same wrapper in the `performUploadRequest` generic? Either something's needs improvement with the naming or this can be further simplified.